masq_lib/src/blockchains/chains.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use crate::blockchains::blockchain_records::{BlockchainRecord, CHAINS};
+use crate::constants::DEFAULT_CHAIN;
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub enum Chain {
+    EthMainnet,
+    EthRopsten,
+    PolyMainnet,
+    PolyMumbai,
+    Dev,
+}
+
+impl Default for Chain {
+    fn default() -> Self {
+        DEFAULT_CHAIN
+    }
+}
+
+impl From<&str> for Chain {
+    fn from(str: &str) -> Self {
+        match str {
+            "polygon-mainnet" => Chain::PolyMainnet,
+            "eth-mainnet" => Chain::EthMainnet,
```

Comment:
`polygon` is not abbreviated yet `eth` is. Does this feel too asymmetrical? Should it be `poly` and `eth` or `polygon` and `ethereum` for symmetry? I also wonder if having `eth-ropsten` and `polygon-mumbai` in the descriptor will mislead people to try to use those as command line parameters. The `from` function can just as easily accept "poly-mainnet", "poly-mumbai", "polygon-mumbai", "eth-ropsten", "ropsten", "eth-mainnet", "mainnet".

Reply:
Could be... but I'd have to take care of clap's App configuration to accept these including the alternative.

Please, I'd really like to hear advice regarding what the look should be like...I can program it but there different preferences of everybody from us. 
Actually I don't care so much...

Now I quite like the idea of short and long version. We could do that.

But as for the abbreviated polygon to just "pol"y; I don't think it's right. If I look at their sites etc I always see them using the full name and so I'd consider it disrespectful to them to mispronounce their name - that's my insight. 
Also...Polygon is not as broadly known as Ethereum is and so I think it's in place to use the full name to help first-contact people understand. On the other hand if I use "eth" I suppose everybody who touches crypto must know. 
That's my reasoning - and even though I'm aware of the asymmetry I would hold it.
Plus another argument - first I claim I didn't take their idea and use it - I noticed later with Alchemy that they resolved the issue the same way as me. They also use "polygon-mumbai" and "eth-ropsten" iand etc. in their URLs.

Reply:
@kauri-hero Please add your opinion.

Reply:
Thanks @bertllll 
thinking more about it all, I think I agree more with your thoughts of ETH being an understand term as abbreviation fo ETH.

I think we should keep eth-mainnet, eth-ropsten, polygon-mainnet & polygon-mumbai for the chain names and descriptor formats.

do you think we should make a card to possibly change the `--chain` acceptable values to be in line with this?
Currently the `--chain` values accepted are:
`mainnet`
`ropsten`
`polygon`
`mumbai`

I think we can live without this alignment, but we need to be clear in our documentation about `--chain` and the fact that descriptors will appear different from the commandline visible `--chain` value that appears in CONFIG table and `setup`/`configuration` values